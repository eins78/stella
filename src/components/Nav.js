import React from 'react'
import {Link} from 'react-router'

const NavLink = ({to, href, ...restProps}) =>
  <Link {...restProps} to={to || href} activeClassName='active' />

const Nav = ({sitemap}) => (
  <nav id='primary'>
    <ol>
      {sitemap.map(([key, page]) =>
        <li key={key}>
          <NavLink href={'/' + key}>{key}</NavLink>
          {/* <ol>
            <li>
              <a href='../../works/insert-coin/'>insert coin</a>
            </li>
          </ol> */}
        </li>
      )}
    </ol>
  </nav>
)

export default Nav
