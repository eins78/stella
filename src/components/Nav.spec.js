/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Nav from './Nav'

it('renders', () => {
  const component = renderer.create(
    <Nav sitemap={[['one', {}], ['two', {}]]} />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
