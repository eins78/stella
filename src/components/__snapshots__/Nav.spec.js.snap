exports[`test renders 1`] = `
<nav
  id="primary">
  <ol>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        one
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        two
      </a>
    </li>
  </ol>
</nav>
`;
